scripts/source-maintenance.sh

```diff
@@ -85,27 +102,38 @@ done
 # an error code seen by a KeepGoing-aware command (or zero)
 SeenErrors=0
 
-
 if ! git diff --quiet; then
 	echo "There are unstaged changes. This script may modify sources."
 	echo "Stage changes to avoid permanent losses when things go bad."
 	exit 1
 fi
 
-# On squid-cache.org we have to use the python scripted md5sum
-HOST=`hostname`
-if test "$HOST" = "squid-cache.org" ; then
-	MD5="md5"
+for Checksum in md5sum md5 shasum sha1sum
+do
+    if "$Checksum" </dev/null >/dev/null 2>/dev/null ; then
+        break
+    fi
+done
+echo "detected checksum program $Checksum"
+
+if [ "x$ASTYLE" != "x" ] ; then
+    if ${ASTYLE} --version >/dev/null 2>/dev/null ; then
+        :
+    else
+        echo "ERROR: cannot run user-supplied astyle ${ASTYLE}"
+    fi
 else
-	MD5="md5sum"
+    for AttemptedBinary in astyle-${TargetAstyleVersion} astyle
```

Comment:
astyle without version suffix is the common case. It should be checked first.

```suggestion
    for AttemptedBinary in astyle astyle-${TargetAstyleVersion}
```

Reply:
There is a much bigger problem here that should probably be addressed first. If we go down this "let's find astyle we can use" route, then we have to find the astyle binary _providing the right astyle version_, not just any astyle binary. Once we do that, the order of checks becomes unimportant (and checking the more popular naming variant as suggested by Amos will start to make sense as opposed to breaking the intent).

Reply:
Are you saying we should list the entire expanded set of PATH permutations for each name  to find any alternative astyle binaries?

Reply:
> Are you saying we should list the entire expanded set of PATH permutations for each name to find any alternative astyle binaries?

I am not. The PR checks two binaries. I am not suggesting to check more (or different) binaries. I am warning Francesco that while the suggested order optimization sounds good, the suggested change must not be applied "as is".

Given the current PR code, the current order of binaries cannot be changed -- applying the suggested change will introduce another bug: Folks that have both `astyle-vGood` and `astyle` (vBad) installed will end up using bad `astyle` while they obviously should use `astyle-vGood`.

To fix this and related problems, we need to change PR code inside this `for` loop. The loop should find a runnable  astyle binary having the expected version (instead of just finding a runnable astyle binary).

Once the check is fixed, the order of listed binaries becomes unimportant from the outcome correctness point of view, and, hence, the suggested order optimization can be applied without breaking anything.

Reply:
Ah. I see. Thanks. Removing this request.